GVFS/GVFS.Common/FileSystem/IVolumeStateWatcher.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace GVFS.Common.FileSystem
+{
+    /// <summary>
+    /// Monitors the system for changes to the state of any disk volume, notifying subscribers when a change occurs.
+    /// </summary>
+    public interface IVolumeStateWatcher : IDisposable

```

Comment:
Would you imagine this to be a 'fire once/self unregister' callback? Or return some cookie/handle as part of the registration process to be used when unregistering otherwise.

Reply:
Fire once makes the most sense to me, but I've realized that we just have different assumptions about what this feature is trying to do. See more on my upcoming note.